ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFExceptionInVertex.java

```diff
@@ -0,0 +1,156 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.udf.generic;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.hadoop.hive.ql.exec.Description;
+import org.apache.hadoop.hive.ql.exec.MapredContext;
+import org.apache.hadoop.hive.ql.exec.UDFArgumentException;
+import org.apache.hadoop.hive.ql.exec.UDFArgumentTypeException;
+import org.apache.hadoop.hive.ql.exec.tez.TezProcessor;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantIntObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantStringObjectInspector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class implements the UDF which can throw an exception in arbitrary vertex (typically mapper)
+ * / task / task attempt. For throwing exception in reducer side, where most probably
+ * GroupByOperator codepath applies, GenericUDAFExceptionInVertex is used.
+ */
+@Description(name = "exception_in_vertex_udf", value = "_FUNC_(vertexName, taskNumberExpression, taskAttemptNumberExpression)")
+public class GenericUDFExceptionInVertex extends GenericUDF {
+  private static final Logger LOG = LoggerFactory.getLogger(GenericUDFExceptionInVertex.class);
+
+  private String vertexName;
+  private String taskNumberExpr;
+  private String taskAttemptNumberExpr;
+  private String currentVertexName;
+  private int currentTaskNumber;
+  private int currentTaskAttemptNumber;
+  private boolean alreadyCheckedAndPassed;
+
+  @Override
+  public ObjectInspector initialize(ObjectInspector[] parameters) throws UDFArgumentException {
+    if (parameters.length < 2) {
+      throw new UDFArgumentTypeException(-1,
+          "At least two argument is expected (fake column ref, vertex name)");
+    }
+
+    this.vertexName = getVertexName(parameters, 1);
+    this.taskNumberExpr = getTaskNumber(parameters, 2);
+    this.taskAttemptNumberExpr = getTaskAttemptNumber(parameters, 3);
```

Comment:
I know it will be mostly just us using this - but it would be helpfull to document the accepted format (and probably throw an exception if something else is passed)

Reply:
thanks for taking a look @kgyrtkirk! there are examples in exception_in_vertex_udf.q, do you think it's enough to mention it here?

Reply:
I was thinking to add some notes about it to the `@Description` - I would look there first in an UDF

Reply:
sure, fixed in new commit